eth/protocols/snap/sync_test.go

```diff
@@ -1752,6 +1795,38 @@ func makeBoundaryStorageTrie(owner common.Hash, n int, db *trie.Database) (commo
 	return root, nodes, entries
 }
 
+// makeUnevenStorageTrie constructs a storage tries will states distributed in
+// different range unevenly.
+func makeUnevenStorageTrie(owner common.Hash, slots int, db *trie.Database) (common.Hash, *trienode.NodeSet, []*kv) {
+	var (
+		entries []*kv
+		tr, _   = trie.New(trie.StorageTrieID(types.EmptyRootHash, owner, types.EmptyRootHash), db)
+		chosen  = make(map[byte]struct{})
+	)
+	for i := 0; i < 3; i++ {
+		var n int
+		for {
+			n = mrand.Intn(15) // the last range is set empty deliberately
+			if _, ok := chosen[byte(n)]; ok {
+				continue
+			}
+			chosen[byte(n)] = struct{}{}
+			break
+		}
+		for j := 0; j < slots/3; j++ {
+			key := append([]byte{byte(n)}, testutil.RandBytes(31)...)
+			val, _ := rlp.EncodeToBytes(testutil.RandBytes(32))
+
+			elem := &kv{key, val}
+			tr.MustUpdate(elem.k, elem.v)
+			entries = append(entries, elem)
+		}
+	}
```

Comment:
How about something like this? 
```suggestion
	for i := 0; i < slots; i++ {
		key := testutil.RandBytes(32)
		if key[0] >= 0x80 {
			key[0] -= 0x80
		}
		val, _ := rlp.EncodeToBytes(testutil.RandBytes(32))
		elem := &kv{key, val}
		tr.MustUpdate(elem.k, elem.v)
		entries = append(entries, elem)
	}
```

Reply:
Actually, if I do 
```
               if key[0] >= 0x80 {
                       key[0] = 0x00
               }

```
Then I start to hit the edgecase we want to repro pretty often